libs/design/src/atoms/form/checkbox/checkbox.component.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { Component, DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { ReactiveFormsModule, FormGroup, FormControl, FormArray } from '@angular/forms';
+import { DaffCheckboxModule } from './checkbox.module';
+import { DaffCheckboxComponent } from './checkbox.component';
+
+
+
+@Component({
+  template: `
+    <daff-checkbox value="testValue"></daff-checkbox>
+  `
+})
+class CheckboxWrapperComponent { }
+
+@Component({
+  template: `
+ 
+<daff-checkbox-set [formGroup]="checkboxGroup" name="toppings">
+  <daff-checkbox formControlName="0" value="lettuce">Apple</daff-checkbox>
+  <daff-checkbox formControlName="1" value="tomato">Grape</daff-checkbox>
+  <daff-checkbox formControlName="2" value="pickle">Peach</daff-checkbox>
+</daff-checkbox-set>
+  `
+})
+class CheckboxEmbeddedComponent {
+
+  lettuce = new FormControl();
+  tomato = new FormControl();
+  pickle = new FormControl();
+  checkboxGroup = new FormArray([this.lettuce, this.tomato, this.pickle]);
+  setValue() {
+    this.checkboxGroup.setValue(['lettuce', 'tomato', 'pickle'])
+  }
+  disable() {
+    this.checkboxGroup.disable();
+  }
+}
+
+describe('DaffCheckboxComponent', () => {
+  let checkboxWrapper: CheckboxWrapperComponent;
+  let checkboxEmbedded: CheckboxEmbeddedComponent;
+
+  let component: DaffCheckboxComponent;
+  let embeddedComponent: DaffCheckboxComponent;
+
+  let fixture: ComponentFixture<CheckboxWrapperComponent>;
+  let embeddedFixture: ComponentFixture<CheckboxEmbeddedComponent>;
+
+  let de: DebugElement;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        CheckboxWrapperComponent,
+        CheckboxEmbeddedComponent,
+      ],
+      imports: [
+        ReactiveFormsModule,
+        DaffCheckboxModule
+      ]
+    })
+      .compileComponents();
+  }));
+
+  describe('daff-checkbox', () => {
+
+    describe('within a daff-checkbox-set', () => {
```

Comment:
The spec for working within `daff-checkbox-set` should be in a different file. The tests here `within a daff-checkbox-set` have nothing to do with a `daff-checkbox-set`.